sdk/trace/trace_test.go

```diff
@@ -1977,80 +1922,123 @@ func TestEmptyRecordingSpanDroppedAttributes(t *testing.T) {
 	assert.Equal(t, 0, (&recordingSpan{}).DroppedAttributes())
 }
 
-func TestAddLinkWithInvalidSpanContext(t *testing.T) {
-	te := NewTestExporter()
-	sl := NewSpanLimits()
-	tp := NewTracerProvider(
-		WithSpanLimits(sl),
-		WithSyncer(te),
-		WithResource(resource.Empty()),
-	)
-	span := startSpan(tp, "AddSpanWithInvalidSpanContext")
-	inValidContext := trace.NewSpanContext(trace.SpanContextConfig{
-		TraceID: trace.TraceID([16]byte{}),
-		SpanID:  [8]byte{},
-	})
-	attrs := []attribute.KeyValue{{Key: "k", Value: attribute.StringValue("v")}}
-	span.AddLink(trace.Link{
-		SpanContext: inValidContext,
-		Attributes:  attrs,
-	})
-
-	want := &snapshot{
-		name: "span0",
-		spanContext: trace.NewSpanContext(trace.SpanContextConfig{
-			TraceID:    tid,
-			TraceFlags: 0x1,
-		}),
-		parent:               sc.WithRemote(true),
-		links:                nil,
-		spanKind:             trace.SpanKindInternal,
-		instrumentationScope: instrumentation.Scope{Name: "AddSpanWithInvalidSpanContext"},
-	}
-	got, err := endSpan(te, span)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if diff := cmpDiff(got, want); diff != "" {
-		t.Errorf("AddLinkWithInvalidSpanContext: -got +want %s", diff)
-	}
-}
+func TestSpanAddLink(t *testing.T) {
+	tests := []struct {
+		name        string
+		spanContext trace.SpanContext
+		validate    func(*testing.T, *snapshot)
```

Comment:
```suggestion
		want        *snapshot
```

Could we reuse the `validate` to make it easy to read?